state/mongodb/mongodb.go

```diff
@@ -256,11 +258,15 @@ func (m *MongoDB) doTransaction(sessCtx mongo.SessionContext, operations []state
 	return nil
 }
 
+func getMongoURI(metadata *mongoDBMetadata) string {
+	return fmt.Sprintf(connectionURIFormat, metadata.username, metadata.password, metadata.host, metadata.databaseName, metadata.params)
+}
+
 func getMongoDBClient(metadata *mongoDBMetadata) (*mongo.Client, error) {
 	var uri string
 
 	if metadata.username != "" && metadata.password != "" {
```

Comment:
If username and password is empty (they can be optional), the url will not be generated.

Reply:
Correct. That was the logic in the original implementation and I didn't want to change it.

@shalabhms can advise here.